### DISCUSSION TITLE
BC setting between two computational domains

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#16874] [Q&A Modules: General] [answered]
[BC_setting_between_two_computational_domains.pdf](https://github.com/idaholab/moose/files/5919080/BC_setting_between_two_computational_domains.pdf)

All,
I am trying to set up boundary conditions between two concentration diffusion computational domains which are a function of concentration and time (please see the attached file).
It would be appreciated if you could give me your suggestion(s) for how to set up these BC settings.
Sincerely,
Thomas

**Comment by GiudGiud** [comment, marked as answer]
Hi Thomas

Let's take the Dirichlet BC for c1 in the center for an example:
I think you should create a new class that derives from DirichletBCBase. You can look at the framework/src/bcs/DirichletBC.C file for an example.
Then in that class you add c2 and c3 as coupled variables (named _c2 and _c3 for example)
-add them to the input parameters with params.addCoupledVar("c2", "the concentration of species 2")
-initialize them in the constructor

Then override the computeValue function from DirichletBCBase with your own function that defines the value that c1 should have on the boundary, based on the values of c2, c3

To introduce time as its own variable (and not through c23(t)), use _t. See SinDirichletBC.C for an example. 

EDIT: Yaqi's got a point. Interface fits the use case more than boundary conditions. 

Hope this helps,
Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
I found the way to work. :)

**Comment by YaqiWang** [comment]
I think you can consider `InterfaceKernel`, which essentially imposes boundary conditions weakly (Neumann) for both sides. I doubt directly using Dirichlet BC will work.

**Reply by styyokuda** [reply]
Thank you, YaqiWang.
I found the way to work. :)
